query-engine/driver-adapters/executor/src/qe.ts

```diff
@@ -0,0 +1,76 @@
+import type { DriverAdapter } from "@prisma/driver-adapter-utils";
+import * as napi from "./engines/Library";
+import * as os from "node:os";
+import * as path from "node:path";
+import { fileURLToPath } from "node:url";
+
+const dirname = path.dirname(fileURLToPath(import.meta.url));
+
+export interface QueryEngine {
+  connect(trace: string): Promise<void>;
+  disconnect(trace: string): Promise<void>;
+  query(body: string, trace: string, tx_id?: string): Promise<string>;
+  startTransaction(input: string, trace: string): Promise<string>;
+  commitTransaction(tx_id: string, trace: string): Promise<string>;
+  rollbackTransaction(tx_id: string, trace: string): Promise<string>;
+}
+
+export type QueryLogCallback = (log: string) => void;
+
+export async function initQueryEngine(
+  engineType: "Napi" | "Wasm",
+  adapter: DriverAdapter,
+  datamodel: string,
+  queryLogCallback: QueryLogCallback,
+  debug: (...args: any[]) => void
+): Promise<QueryEngine> {
+  const logCallback = (event: any) => {
+    const parsed = JSON.parse(event);
+    if (parsed.is_query) {
+      queryLogCallback(parsed.query);
+    }
+    debug(parsed);
+  };
+
+  const options = queryEngineOptions(datamodel);
+
+  if (engineType === "Wasm") {
+    const { WasmQueryEngine } = await import("./wasm");
+    return new WasmQueryEngine(options, logCallback, adapter);
+  } else {
+    const { QueryEngine } = loadNapiEngine();
+    return new QueryEngine(options, logCallback, adapter);
+  }
+}
+
+export function queryEngineOptions(datamodel: string) {
+  return {
+    datamodel,
+    configDir: ".",
+    engineProtocol: "json" as const,
+    logLevel: process.env["RUST_LOG"] ?? ("info" as any),
+    logQueries: true,
+    env: process.env,
+    ignoreEnvVarErrors: false,
+  };
+}
+
+function loadNapiEngine(): napi.Library {
+  // I assume nobody will run this on Windows ¯\_(ツ)_/¯
+  const libExt = os.platform() === "darwin" ? "dylib" : "so";
+  const target =
+    process.env.TARGET || process.env.PROFILE == "release"
+      ? "release"
+      : "debug";
+
+  const libQueryEnginePath = path.join(
```

Comment:
```suggestion
  const libQueryEnginePath = path.resolve(
```

If we'll ever want to log and debug it, having canonical path would be much more conveinient.